### PR TITLE
fix: strict dependencies to fix build

### PR DESCRIPTION
**Pull request by abetaev**
hi,

at some point i came up with necessity to build js-libp2p, but found that it's not buildable.

[this change was the cause (libp2p/interfaces)](https://github.com/libp2p/js-libp2p-interfaces/commit/a367375accc690d7b4608c9a3313f91df700efd8)

if semver rules were followed this shouldn't have happened.

### CODE REVIEW

**Comment by achingbrain** [comment]
Thanks for the PR.  The relevant changes needed to support the updated interfaces have been done as part of #1206 so if you pull from master and reinstall the node deps it should build successfully again.

> if semver rules were followed this shouldn't have happened.

This project is in a bit of a weird space while we [transition to TypeScript](https://github.com/libp2p/js-libp2p/issues/1021) - the interfaces have grown organically over time so are subject to a bit of churn while we apply the learnings of the past few years to them so there will be some breakage.  The good news is this breakage is contained to the `@libp2p/*` modules and `libp2p@next` - the previous `0.36.x` release tree is unaffected because it doesn't use and `@libp2p/*` modules.

Long story short: as soon as #1178 is merged and `0.37.x` is released, semver will be strictly followed again.  In the interim you should use the `0.36.x` release (see [releases](https://github.com/libp2p/js-libp2p/releases) for the latest tag) and not `libp2p@next` or `master`.

**Comment by abetaev** [comment]
thank you for the news and the explanation, it makes things clearer
